File: android/app/src/main/java/jagdx/IDirect3D.java
package jagdx;

import jaclib.peer.IUnknown;
import jaclib.peer.ti;

import voidawt.Canvas;

public final class IDirect3D extends IUnknown {

    public final ti b;

    public static int _Direct3DCreate(int arg0, IDirect3D arg1) { return 0; }

    public static IDirect3D a(int arg0, ti arg1) {
        IDirect3D local4 = new IDirect3D(arg1);
        int local8 = _Direct3DCreate(arg0, local4);
        if (lh.a(local8, true)) {
            throw new sja(String.valueOf(local8));
        }
        return local4;
    }

    public IDirect3D(ti arg0) {
        super(arg0);
        this.b = arg0;
    }

    public int CheckDeviceFormat(int arg0, int arg1, int arg2, int arg3, int arg4, int arg5) { return 0; }

    public IDirect3DDevice a(int arg0, int arg1, Canvas arg2, int arg3, D3DPRESENT_PARAMETERS arg4) {
        IDirect3DDevice local5 = new IDirect3DDevice(this.b);
        int local14 = this._CreateDevice(arg0, arg1, arg2, arg3, arg4, local5);
        if (lh.a(local14, true)) {
            throw new sja(String.valueOf(local14));
        }
        return local5;
    }

    public int a(int arg0, D3DDISPLAYMODE arg1) {
        return this._GetAdapterDisplayMode(arg0, arg1);
    }

    public int _GetAdapterDisplayMode(int arg0, D3DDISPLAYMODE arg1) { return 0; }

    public int _GetAdapterIdentifier(int arg0, int arg1, D3DADAPTER_IDENTIFIER arg2) { return 0; }

    public int CheckDepthStencilMatch(int arg0, int arg1, int arg2, int arg3, int arg4) { return 0; }

    public int _GetDeviceCaps(int arg0, int arg1, D3DCAPS arg2) { return 0; }

    public D3DCAPS b(int arg0, int arg1) {
        D3DCAPS local3 = new D3DCAPS();
        int local9 = this._GetDeviceCaps(arg0, arg1, local3);
        if (lh.a(local9, true)) {
            throw new sja(String.valueOf(local9));
        }
        return local3;
    }

    public int CheckDeviceMultiSampleType(int arg0, int arg1, int arg2, boolean arg3, int arg4) { return 0; }

    public int _CreateDevice(int arg0, int arg1, Canvas arg2, int arg3, D3DPRESENT_PARAMETERS arg4, IDirect3DDevice arg5) { return 0; }

    public D3DADAPTER_IDENTIFIER a(int arg0, int arg1) {
        D3DADAPTER_IDENTIFIER local3 = new D3DADAPTER_IDENTIFIER();
        int local9 = this._GetAdapterIdentifier(arg0, arg1, local3);
        if (lh.a(local9, true)) {
            throw new sja(String.valueOf(local9));
        }
        return local3;
    }

    public int CheckDeviceType(int arg0, int arg1, int arg2, int arg3, boolean arg4) { return 0; }
}

File: src/jvmMain/kotlin/jagdx/IDirect3D.kt
package jagdx

import jaclib.peer.IUnknown
import jaclib.peer.hb
import awt.Canvas
import kotlin.jvm.JvmStatic

actual class IDirect3D /*private*/ actual constructor(/*private*/ val b: hb?) : IUnknown(b) {
    actual fun a(arg0: Int, arg1: D3DDISPLAYMODE?): Int {
        return this._GetAdapterDisplayMode(arg0, arg1)
    }

    actual fun a(arg0: Int, arg1: Int, arg2: Canvas?, arg3: Int, arg4: D3DPRESENT_PARAMETERS?): IDirect3DDevice {
        val local5 = IDirect3DDevice(this.b)
        val local14 = this._CreateDevice(arg0, arg1, arg2, arg3, arg4, local5)
        if (ue.a(97.toByte(), local14)) {
            throw fda(local14.toString())
        }
        return local5
    }

    actual fun a(arg0: Int, arg1: Int): D3DADAPTER_IDENTIFIER {
        val local3 = D3DADAPTER_IDENTIFIER()
        val local9 = this._GetAdapterIdentifier(arg0, arg1, local3)
        if (ue.a(97.toByte(), local9)) {
            throw fda(local9.toString())
        }
        return local3
    }

    actual external fun CheckDeviceType(arg0: Int, arg1: Int, arg2: Int, arg3: Int, arg4: Boolean): Int

    actual fun b(arg0: Int, arg1: Int): D3DCAPS {
        val local3 = D3DCAPS()
        val local9 = this._GetDeviceCaps(arg0, arg1, local3)
        if (ue.a(97.toByte(), local9)) {
            throw fda(local9.toString())
        }
        return local3
    }

    actual external fun CheckDeviceMultiSampleType(arg0: Int, arg1: Int, arg2: Int, arg3: Boolean, arg4: Int): Int

    /*private*/ actual external fun _GetAdapterDisplayMode(arg0: Int, arg1: D3DDISPLAYMODE?): Int

    /*private*/ actual external fun _CreateDevice(arg0: Int, arg1: Int, arg2: Canvas?, arg3: Int, arg4: D3DPRESENT_PARAMETERS?, arg5: IDirect3DDevice?): Int

    actual external fun CheckDepthStencilMatch(arg0: Int, arg1: Int, arg2: Int, arg3: Int, arg4: Int): Int

    actual external fun CheckDeviceFormat(arg0: Int, arg1: Int, arg2: Int, arg3: Int, arg4: Int, arg5: Int): Int

    /*private*/ actual external fun _GetAdapterIdentifier(arg0: Int, arg1: Int, arg2: D3DADAPTER_IDENTIFIER?): Int

    /*private*/ actual external fun _GetDeviceCaps(arg0: Int, arg1: Int, arg2: D3DCAPS?): Int

    companion actual object {
        @JvmStatic
        actual fun a(arg0: Int, arg1: hb?): IDirect3D {
            val local4 = IDirect3D(arg1)
            val local8: Int = _Direct3DCreate(arg0, local4)
            if (ue.a(97.toByte(), local8)) {
                throw fda(local8.toString())
            }
            return local4
        }

        @JvmStatic
        /*private*/ actual external fun _Direct3DCreate(arg0: Int, arg1: IDirect3D?): Int
    }
}
